mint/run/core/s3select/csv.py

```diff
@@ -0,0 +1,204 @@
+                            RecordDelimiter="\n",
+                            FieldDelimiter=",",
+                            QuoteCharacter=quote_char,
+                            QuoteEscapeCharacter=escape_char,
+                            Comments="#",
+                            AllowQuotedRecordDelimiter="FALSE",),
+                        ),
+                    output_serialization=OutputSerialization(
+                        json = JsonOutput(
+                           RecordDelimiter="\n",
+                           )
+                        ),
+                    request_progress=RequestProgress(
+                        enabled="False"
+                        )
+                    )
+
+            got_output = b''
+
+            try:
+                got_output = exec_select(client, bucket_name, object_content, options, log_output)
+            except Exception as select_err:
+                if not isinstance(expected_output, Exception):
+                    raise ValueError('Test {} unexpectedly failed with: {}'.format(idx+1, select_err))
+            else:
+                if isinstance(expected_output, Exception):
+                    raise ValueError('Test {}: expected an exception, got {}'.format(idx+1, got_output))
+                if got_output != expected_output:
+                    raise ValueError('Test {}: data mismatch. Expected : {}, Received {}'.format(idx+1, expected_output, got_output))
+
+    except Exception as err:
+        raise Exception(err)
+    finally:
+        try:
+            client.remove_bucket(bucket_name)
+        except Exception as err:
+            raise Exception(err)
+
+    # Test passes
+    print(log_output.json_report())
+
+def test_csv_output_custom_quote_char(client, log_output):
+    # Get a unique bucket_name and object_name
+    log_output.args['bucket_name'] = bucket_name = generate_bucket_name()
+
+    tests = [
+            # UTF-8 quote character
+            ("''", "''", b'col1,col2,col3\n', Exception()),
+            ("'", "'", b'col1,col2,col3\n', b"'col1','col2','col3'\n"),
+            ("", '"', b'col1,col2,col3\n', b'\x00col1\x00,\x00col2\x00,\x00col3\x00\n'),
+            ('"', '"', b'col1,col2,col3\n', b'"col1","col2","col3"\n'),
+            ('"', '"', b'col"1,col2,col3\n', b'"col""1","col2","col3"\n'),
+            ('"', '"', b'""""\n', b'""""\n'),
+            ('"', '"', b'\n', b''),
+            ("'", "\\", b'col1,col2,col3\n', b"'col1','col2','col3'\n"),
+            ("'", "\\", b'col""1,col2,col3\n', b"'col\"\"1','col2','col3'\n"),
+            ("'", "\\", b'col\'1,col2,col3\n', b"'col\\'1','col2','col3'\n"),
+            ("'", "\\", b'"col\'1","col2","col3"\n', b"'col\\'1','col2','col3'\n"),
+            ("'", "\\", b'col\'\n', b"'col\\''\n"),
+            # Two consecutive escaped quotes
+            ("'", "\\", b'"a"""""\n', b"'a\"\"'\n"),
+    ]
+
+    try:
+        client.make_bucket(bucket_name)
```

Comment:
1. Can be used like above comment.
2. As this function repeating the above function's logic, we could have a common function and passing appropriate argument to archive the same.

Reply:
> * Can be used like above comment.

Done

> 2\. As this function repeating the above function's logic, we could have a common function and passing appropriate argument to archive the same.

Not sure, this function is only long because of sql input/output options

Reply:
> Not sure, this function is only long because of sql input/output options

Changed my mind, will do it

Reply:
> 2\. As this function repeating the above function's logic, we could have a common function and passing appropriate argument to archive the same.

Done